### ISSUE TITLE
The Viber software has been updated, but when opened using the sandboxie program, the system program always crashes.

### ISSUE THREAD

**Issue by dingfeng78900** [sandboxie-plus/Sandboxie#4298]
### Describe what you noticed and did

1.Download the latest version of Viber on your computer system.
2.Opening the application in sandboxie shows that the sandboxie program crashes after running for a few seconds.

### How often did you encounter it so far?

every time

### Expected behavior

Please resolve this issue as soon as possible.

### Affected program

Not relevant

### Download link

https://www.viber.com/en/download/

### Where is the program located?

I tried to install it only inside a sandbox, but I wasn't able to achieve it.

### Did the program or any related process close unexpectedly?

No, not at all.

### What version of Sandboxie are you running now?

Sandboxie-Plus-x64-v1.14.10.exe 64 Windows 10 

### Is it a new installation of Sandboxie?

I recently did a new clean installation.

### Is it a regression from previous versions?

Sandboxie-Plus-x64-v1.14.10

### In which sandbox type you have this problem?

In a standard isolation sandbox (yellow sandbox icon).

### Can you reproduce this problem on a new empty sandbox?

I can confirm it also on a new empty sandbox.

### What is your Windows edition and version?

Windows 10 Pro 22H2 64-bit

### In which Windows account you have this problem?

A local account (Standard user).

### Please mention any installed security software

No security software

### Did you previously enable some security policy settings outside Sandboxie?

No security software

### Sandboxie.ini configuration

```ini
Not relevant
```

**Comment by dingfeng78900** [comment]
The old version of the Viber application can be opened normally. Viber was updated on October 9, 2024. The updated Viber will show a program crash after opening in sandboxie.

**Comment by offhub** [comment]
[Viber.exe.775987882.dmp.7z.zip](https://github.com/user-attachments/files/17381388/Viber.exe.775987882.dmp.7z.zip)

**Comment by xhienchi** [comment]
same as problem for me

**Comment by Djknjaz** [comment]
I have the same problem. Help please. I'm adding a dump.
[dmp.zip](https://github.com/user-attachments/files/17412282/dmp.zip)

**Comment by dsfdsf34** [comment]
I agree with you, the latest version of Viber that works in the sandbox is 23.3.0.0, but it is no longer supported by the developer for account activation

**Comment by dsfdsf34** [comment]
> Viber.exe works without problems when launched through the debugger, but crashes when launched normally.
> 
> [Viber.exe.775987882.dmp.7z.zip](https://github.com/user-attachments/files/17381388/Viber.exe.775987882.dmp.7z.zip)

How can I run Viber through the debugger so that it works?

**Comment by dsfdsf34** [comment]
There is a solution, use the sandbox type - container for applications, Viber does not work with the sandbox with standard isolation.
But for this function you need to buy a subscription to the sandbox if you want your window to work longer than 5 minutes

**Comment by PizzaProgram** [comment]
I've just installed Sandboxie-Plus for only one reason:  to install Viber.
The installation "stops" at 100%, nothing happens. _(Well, the log counter of Sandboxie is growning by ca +20/sec)_
After 30 minutes of waiting I gave up and after some search, found this topic.

Is there any progress to install + Run Viber in normal isolation mode?

**Comment by dsfdsf34** [comment]
> I've just installed Sandboxie-Plus for only one reason: to install Viber. The installation "stops" at 100%, nothing happens. _(Well, the log counter of Sandboxie is growning by ca +20/sec)_ After 30 minutes of waiting I gave up and after some search, found this topic.
> 
> Is there any progress to install + Run Viber in normal isolation mode?


You are doing it wrong.

1. Install Viber on your computer, and after installation, close it immediately, no need to log into your account

2. Create a sandbox, specifically "container for applications" it is green, open the shortcut of the installed Viber on your PC, log into your account in this sandbox.

**Comment by PizzaProgram** [comment]
@dsfdsf34 Thanks for trying to help, but:

1. If you install Viber on your PC, it already infected it with thousands of Registry tweaks, DLLs in the Win\system32 dir, and who knows what else... _(There are 100K+ log lines what is does during install. Tried to analyse 30min long, but it's too much and SB's log is not detailed enough.)_

2.  The green  "application container" is only available for users who **pay** for this "free" software.

**Comment by dsfdsf34** [comment]
@PizzaProgram Thanks for the reply, I've been using Viber for a long time and...
this is the only working way to make Viber work in a sandbox so you can open multiple accounts at the same time on one PC.

If you're so worried about what Viber does during installation, then use a virtual machine and "buy" a sandbox key so you can create a green "app container"

**Comment by offhub** [comment]
### Using a Shortcut to Launch Viber in Sandboxie Plus

1. **Enable Keep Terminated Setting**  
   In Sandboxie Plus, go to the `View` menu and enable the `Keep Terminated` setting.

2. **Run Viber in the Sandbox**  
   Launch Viber inside the sandbox.

3. **Create a Shortcut for Viber**  
   - Right-click on `Viber.exe` in the sandbox program list.  
   - Select the option to `Create Shortcut`.

4. **Edit the Shortcut Properties**  
   Open the properties of the newly created shortcut.  

5. **Modify the Target Section**  
   Replace the `Target` section of the shortcut with the following command, after replacing the `<BoxName>` with the name of the sandbox you are using:
   ```cmd
   "C:\Program Files\Sandboxie-Plus\Start.exe" /env:QTWEBENGINE_DISABLE_SANDBOX=1 /box:<BoxName> "%LocalAppData%\Viber\Viber.exe"
   ```

> [!WARNING]
> Disabling the Qt WebEngine sandbox reduces the security of the application. Use caution.

6. **Run the Shortcut**  
   Launch Viber using the shortcut you created to see if it works correctly.

---

### Alternative Method: Using Triggers in Sandboxie Plus

#### **Using `StartProgram` Trigger**  
The `StartProgram` trigger activates when the first program starts in the sandbox.

> [!TIP]
> Use this trigger if Viber is installed within the sandbox, or run the following command once inside the sandbox if you are not deleting the sandbox contents.

1. In Sandboxie Plus, navigate to `Sandbox Options > Advanced Options > Triggers`.  
2. Create a trigger under `On Box Start: Run Command` with the following command:
    ```cmd
    REG ADD "HKCU\Environment" /v QTWEBENGINE_DISABLE_SANDBOX /t REG_SZ /d 1 /f
    ```

> [!WARNING]
> Disabling the Qt WebEngine sandbox reduces the security of the application. Use caution.

#### **Using `AutoExec` Trigger**  
The `AutoExec` trigger runs every time the sandbox is initialized.

> [!TIP]
> Use this trigger if your sandbox is automatically deleted after each use.

1. Navigate to `Sandbox Options > Advanced Options > Triggers`.  
2. Create a trigger under `On Box Init: Run Command` with this command:
    ```cmd
    REG ADD "HKCU\Environment" /v QTWEBENGINE_DISABLE_SANDBOX /t REG_SZ /d 1 /f
    ```

> [!WARNING]
> Disabling the Qt WebEngine sandbox reduces the security of the application. Use caution.

--- 

#### To install Viber app in the sandbox, you can use the following settings.

```ini
MsiInstallerExemptions=y
ClosedFilePath=msiexec.exe,*\ie4uinit.exe
```

> [!Note]
> If you experience a crash while starting after the initial installation, restart the Viber application.
